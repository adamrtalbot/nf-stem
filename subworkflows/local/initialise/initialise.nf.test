nextflow_pipeline {

    name "Test Subworkflow INITIALISE"
    script "subworkflows/local/initialise/main.nf"
    // workflow 'INITIALISE'

    test("Should run without failures") {

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should show help") {
        when {
            params {
                help = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should show version") {
        when {
            params {
                version = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match()
        }

    }

    test("Should fail because no params.input") {
        when {
            params {
                input = null
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus == 1
            assert snapshot(workflow).match()
        }

    }

}
