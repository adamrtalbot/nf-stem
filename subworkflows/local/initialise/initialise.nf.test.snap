{
    "Should show help": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "Could not read parameters settings from JSON. java.nio.file.NoSuchFileException: /Users/adam.talbot/nf-stem/subworkflows/local/initialise/nextflow_schema.json"
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-09-12T12:10:53.526692"
    },
    "Should fail because no params.input": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 1,
                "failed": true,
                "stdout": [
                    "Please provide an input samplesheet to the pipeline e.g. '--input samplesheet.csv'"
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": false
            }
        ],
        "timestamp": "2023-09-12T12:10:57.376616"
    },
    "Should run without failures": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "WARN: There's no process matching config selector: FASTP",
                    "WARN: There's no process matching config selector: FASTQC",
                    "WARN: There's no process matching config selector: TRIMGALORE"
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-09-12T12:10:51.5897"
    },
    "Should show version": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 0,
                    "tasksSucceeded": 0
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-09-12T12:10:55.47093"
    }
}